src/test/java/ExportTest.java

```diff
@@ -48,17 +53,35 @@ public static boolean isGraphvizInstalled() throws InterruptedException {
     @Test
     @EnabledIf("isGraphvizInstalled")
     public void export() throws IOException {
+        var testCasePath = RESOURCE_DIR.resolve("testcase.lg");
+        var actualPath = RESOURCE_DIR.resolve("actual.tex");
+        var expectedPath = RESOURCE_DIR.resolve("expected.tex");
+
         // Deserialize the test case.
-        var testFile = Path.of("src/test/resources/serialize/testcase.lg");
         DiffTree diffTree;
-        try (BufferedReader lineGraph = Files.newBufferedReader(testFile)) {
-            diffTree = LineGraphImport.fromLineGraph(lineGraph, testFile, importOptions).get(0);
+        try (BufferedReader lineGraph = Files.newBufferedReader(testCasePath)) {
+            diffTree = LineGraphImport.fromLineGraph(lineGraph, testCasePath, importOptions).get(0);
         }
 
         // Export the test case
-        var tikzOutput = new ByteArrayOutputStream();
-        new TikzExporter(format).exportDiffTree(diffTree, tikzOutput);
+        try (
+                var unbufferedOutput = Files.newOutputStream(actualPath);
+                var output = new BufferedOutputStream(unbufferedOutput)
+        ) {
+            new TikzExporter(format).exportDiffTree(diffTree, output);
+        }
 
-        TestUtils.assertEqualToFile(Path.of("src/test/resources/serialize/expected.tex"), tikzOutput.toString());
+        try (
+                var expectedFile = Files.newBufferedReader(expectedPath);
+                var actualFile = Files.newBufferedReader(actualPath);
+        ) {
+            if (!IOUtils.contentEqualsIgnoreEOL(expectedFile, actualFile)) {
+                fail("The DiffTree in file " + testCasePath + " didn't parse correctly. "
+                    + "Expected the content of " + expectedPath + " but got the content of " + actualPath + ". ");
+            } else {
+                // Keep output file for debugging on errors
```

Comment:
Comments is a bit confusing as is right now because it says "keep file" directly in front of "Files.delete". Please adapt to "delete files of successful tests" or something like that.

Reply:
You're right. There are now two comments explaining the action in both branches.